### PR TITLE
Engine cleanup & formatting

### PR DESCRIPTION
**Pull request by W2Wizard**
# Oh god what have you done?

Lots lots and lots of code cleanup via code analysis.

Overall Its applying 3 main improvements:
- `Variable declaration can be inlined`
- `Object initialization can be simplified`
- `Use compount statement`

Also some general formatting making things a bit spaced out since theres a few parts that just feel too dense.

### CODE REVIEW

**Review by stefnotch — CHANGES_REQUESTED** [review]

**Comment by stefnotch** [comment]
Okay, I took a look at the entire thing and haven't found anything besides the comments above.

File: Source/Editor/GUI/Timeline/Tracks/ParticleEmitterTrack.cs [hunk 1]
```diff
@@ -68,7 +68,6 @@ private static void LoadTrack(int version, Track track, BinaryReader stream)
             var e = (ParticleEmitterTrack)track;
             Guid id = new Guid(stream.ReadBytes(16));
             e.Asset = FlaxEngine.Content.LoadAsync<ParticleEmitter>(id);
-            var emitterIndex = stream.ReadInt32();
```
[the remark this answers is not in the document]
Comment (W2Wizard) [reply Source/Editor/GUI/Timeline/Tracks/ParticleEmitterTrack.cs:71]: Fixed

File: Source/Editor/Options/OptionsModule.cs [hunk 2]
```diff
@@ -212,45 +212,46 @@ private void SetupStyle()
...
+                Cross = Editor.Icons.Cross12,
+                CheckBoxIntermediate = Editor.Icons.CheckBoxIntermediate12,
+                CheckBoxTick = Editor.Icons.CheckBoxTick12,
+                StatusBarSizeGrip = Editor.Icons.StatusBarSizeGrip12,
+                Translate = Editor.Icons.Translate16,
+                Rotate = Editor.Icons.Rotate16,
+                Scale = Editor.Icons.Scale16
+            };
             style.DragWindow = style.BackgroundSelected * 0.7f;
             style.ProgressNormal = Color.FromBgra(0xFF0ad328);
```
[the remark this answers is not in the document]
Comment (W2Wizard) [reply Source/Editor/Options/OptionsModule.cs]: Done

File: Source/Editor/SceneGraph/GUI/ActorTreeNode.cs [hunk 3]
```diff
@@ -386,15 +385,20 @@ public ReparentAction(Actor actor)
 
             public ReparentAction(List<Actor> actors)
             {
-                var allActors = new List<Actor>();
-                allActors.Capacity = Mathf.NextPowerOfTwo(actors.Count);
+                var allActors = new List<Actor>
+                {
+                    Capacity = Mathf.NextPowerOfTwo(actors.Count)
+                };
+
```
[the remark this answers is not in the document]
Comment (W2Wizard) [reply Source/Editor/SceneGraph/GUI/ActorTreeNode.cs:389]: Oops!

File: Source/Editor/SceneGraph/GUI/ActorTreeNode.cs [hunk 4]
```diff
@@ -386,15 +385,20 @@ public ReparentAction(Actor actor)
...
                 {
                     GetAllActors(allActors, actors[i]);
                 }
 
-                var allScripts = new List<Script>();
-                allScripts.Capacity = allActors.Capacity;
+                var allScripts = new List<Script>
+                {
+                    Capacity = allActors.Capacity
+                };
```
[the remark this answers is not in the document]
Comment (W2Wizard) [reply Source/Editor/SceneGraph/GUI/ActorTreeNode.cs]: Done

File: Source/Editor/Gizmo/TransformGizmoBase.Selection.cs [hunk 5]
```diff
@@ -56,104 +56,107 @@ private void SelectAxis()
...
-                    _activeAxis = Axis.X;
-                    closestintersection = intersection;
-                }
-                if (YAxisBox.Intersects(ref localRay, out intersection) && intersection < closestintersection)
-                {
-                    _activeAxis = Axis.Y;
-                    closestintersection = intersection;
-                }
-                if (ZAxisBox.Intersects(ref localRay, out intersection) && intersection < closestintersection)
+                case Mode.Translate:
```
Comment (stefnotch) [inline Source/Editor/Gizmo/TransformGizmoBase.Selection.cs:59]: I think the "missing" indent for the `case` labels is intentional.

**Review by stefnotch — COMMENTED** [review]
Looks good!

File: Source/Editor/Gizmo/TransformGizmoBase.Selection.cs (hunk 5 above)
Comment (mafiesto4) [reply Source/Editor/Gizmo/TransformGizmoBase.Selection.cs:59]: Yeah we don't indent cases in switches.

File: Source/Editor/Viewport/PrefabWindowViewport.cs [hunk 6]
```diff
@@ -730,8 +749,7 @@ private void Spawn(Actor actor)
 
         private void Spawn(ScriptType item, SceneGraphNode hit, ref Vector3 hitLocation)
         {
-            var actor = item.CreateInstance() as Actor;
-            if (actor == null)
+            if (!(item.CreateInstance() is Actor actor))
```
Comment (mafiesto4) [inline Source/Editor/Viewport/PrefabWindowViewport.cs:752]: The previous code is more explicit about object creation and I think it should stay.

**Review by mafiesto4 — CHANGES_REQUESTED** [review]

**Review by mafiesto4 — APPROVED** [review]
LGTM

**Pull request merged**